openshift/tests-extension/test/olmv1-incompatible.go

```diff
@@ -0,0 +1,500 @@
+				"name": name,
+			},
+		},
+	}
+}
+
+func createBuildConfig(name, namespace string) *buildv1.BuildConfig {
+	return &buildv1.BuildConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"name": name,
+			},
+		},
+		Spec: buildv1.BuildConfigSpec{
+			CommonSpec: buildv1.CommonSpec{
+				Source: buildv1.BuildSource{
+					Type: buildv1.BuildSourceBinary,
+				},
+				Strategy: buildv1.BuildStrategy{
+					Type: buildv1.DockerBuildStrategyType,
+					DockerStrategy: &buildv1.DockerBuildStrategy{
+						ForcePull: true,
+						From: &corev1.ObjectReference{
+							Kind: "DockerImage",
+							Name: "scratch",
+						},
+						Env: []corev1.EnvVar{
+							{
+								Name:  "BUILD_LOGLEVEL",
+								Value: "5",
+							},
+						},
+					},
+				},
+				Output: buildv1.BuildOutput{
+					To: &corev1.ObjectReference{
+						Kind: "ImageStreamTag",
+						Name: name + ":latest",
+					},
+				},
+			},
+		},
+	}
+}
+
+func waitForBuildToFinish(ctx SpecContext, name, namespace string) {
+	const typeBuildConditionComplete = "Complete"
+	k8sClient := env.Get().K8sClient
+	Eventually(func(g Gomega) {
+		b := &buildv1.Build{}
+		err := k8sClient.Get(ctx, client.ObjectKey{Name: name, Namespace: namespace}, b)
+		g.Expect(err).ToNot(HaveOccurred())
+
+		conditions := b.Status.Conditions
+		var cond *buildv1.BuildCondition
+		for i := range conditions {
+			if conditions[i].Type == typeBuildConditionComplete {
+				cond = &conditions[i]
+				break
+			}
+		}
+		g.Expect(cond).ToNot(BeNil())
+		g.Expect(cond.Status).To(Equal(corev1.ConditionTrue))
+	}).WithTimeout(5 * time.Minute).WithPolling(1 * time.Second).Should(Succeed())
+}
+
+func waitForClusterCatalogServing(ctx SpecContext, name string) {
+	k8sClient := env.Get().K8sClient
+	Eventually(func(g Gomega) {
+		cc := &olmv1.ClusterCatalog{}
+		err := k8sClient.Get(ctx, client.ObjectKey{Name: name}, cc)
+		g.Expect(err).ToNot(HaveOccurred())
+
+		serving := meta.FindStatusCondition(cc.Status.Conditions, olmv1.TypeServing)
+		g.Expect(serving).ToNot(BeNil())
+		g.Expect(serving.Status).To(Equal(metav1.ConditionTrue))
+	}).WithTimeout(5 * time.Minute).WithPolling(1 * time.Second).Should(Succeed())
+}
+
+func waitForClusterExtensionInstalled(ctx SpecContext, name string) {
+	k8sClient := env.Get().K8sClient
+	Eventually(func(g Gomega) {
+		ce := &olmv1.ClusterExtension{}
+		err := k8sClient.Get(ctx, client.ObjectKey{Name: name}, ce)
+		g.Expect(err).ToNot(HaveOccurred())
+
+		progressing := meta.FindStatusCondition(ce.Status.Conditions, olmv1.TypeProgressing)
+		g.Expect(progressing).ToNot(BeNil())
+		g.Expect(progressing.Status).To(Equal(metav1.ConditionTrue))
+
+		installed := meta.FindStatusCondition(ce.Status.Conditions, olmv1.TypeInstalled)
+		g.Expect(installed).ToNot(BeNil())
+		g.Expect(installed.Status).To(Equal(metav1.ConditionTrue))
+	}).WithTimeout(5 * time.Minute).WithPolling(1 * time.Second).Should(Succeed())
+}
```

Comment:
Could we use instead:

https://github.com/openshift/operator-framework-operator-controller/blob/56f527af1025f3e1d5ff46dd58614b25cf6501ca/openshift/tests-extension/pkg/helpers/cluster_extension.go#L113-L131

So, we keep those centralised.

Reply:
Yes, I modified the code to use this.